var/spack/repos/builtin/packages/hiop/package.py

```diff
@@ -6,50 +6,156 @@
+    variant(
+        'sparse',
+        default=False,
+        description='Enable/Disable Sparse linear algebra')
+    variant('deepchecking', default=False,
+            description='Ultra safety checks - '
+            'used for increased robustness and self-diagnostics')
 
-    depends_on('mpi', when='+mpi')
     depends_on('lapack')
     depends_on('blas')
+    depends_on('cmake@3.18:', type='build')
+
+    depends_on('mpi', when='+mpi')
+
+    depends_on('magma', when='+cuda')
+    depends_on('magma@2.5.4', when='@0.3.99.1:+cuda')
+
+    depends_on('raja', when='+raja')
+    depends_on('umpire', when='+raja')
+
+    depends_on('suite-sparse', when='+kron')
+
+    depends_on('coinhsl', when='+sparse')
+    depends_on('metis', when='+sparse')
 
     flag_handler = build_system_flags
 
     def cmake_args(self):
         args = []
         spec = self.spec
 
-        if '+mpi' in spec:
-            args.append("-DWITH_MPI=ON")
-        else:
-            args.append("-DWITH_MPI=OFF")
-
-        if '+deepchecking' in spec:
-            args.append("-DDEEP_CHECKING=ON")
-        else:
-            args.append("-DDEEP_CHECKING=OFF")
-
         lapack_blas_libs = (
             spec['lapack'].libs + spec['blas'].libs).joined(';')
         args.extend([
             '-DLAPACK_FOUND=TRUE',
             '-DLAPACK_LIBRARIES={0}'.format(lapack_blas_libs)
         ])
 
+        args.append('-DHIOP_BUILD_STATIC=ON')
+        if '+shared' in spec:
+            args.append('-DHIOP_BUILD_SHARED=ON')
+        else:
+            args.append('-DHIOP_BUILD_SHARED=OFF')
+
+        if '+mpi' in spec:
+            args.append(
+                "-DHIOP_USE_MPI=ON -DMPI_HOME={0}".format(spec['mpi'].prefix))
+            args.append('-DMPI_C_COMPILER={0}'.format(spec['mpi'].mpicc))
+            args.append('-DMPI_CXX_COMPILER={0}'.format(spec['mpi'].mpicxx))
+            args.append('-DMPI_Fortran_COMPILER={0}'.format(spec['mpi'].mpifc))
+        else:
+            args.append("-DHIOP_USE_MPI=OFF")
+
+        if '+deepchecking' in spec:
+            args.append("-DHIOP_DEEPCHECKS=ON")
+        else:
+            args.append("-DHIOP_DEEPCHECKS=OFF")
+
+        # HIP flags are a part of the buildsystem, but full support is not
+        # yet ready for public release
+        args.append("-DHIOP_USE_HIP=OFF")
+
+        if '+cuda' in spec:
+            args.append("-DHIOP_USE_GPU=ON")
+
+            cuda_arch_list = spec.variants['cuda_arch'].value
+            cuda_arch = cuda_arch_list[0]
+            if cuda_arch != 'none':
+                args.append("-DHIOP_NVCC_ARCH=sm_{0}".format(cuda_arch))
+                args.append("-DCMAKE_CUDA_ARCHITECTURES={0}".format(cuda_arch))
+            args.append("-DHIOP_USE_CUDA=ON")
+
+            args.append('-DHIOP_USE_MAGMA=ON')
+            if '+magma' in spec:
+                args.append(
+                    "-DHIOP_MAGMA_DIR={0}".format(spec['magma'].prefix))
```

Comment:
Are all these `*_DIR` necessary? Spack sets `CMAKE_PREFIX_PATH` to all dependency prefixes for you, so if you have a cmake scripts that runs `find_*` it should be automatically found.

Reply:
Not all of our dependencies use cmake and we roll many of our own `find_*.cmake`, so removing these variables makes me a bit nervous. I'm open to testing this out down the line - would you be open to my team addressing with this in another PR when I have time to test it out?

Reply:
Yeah, let's do that in another PR! Don't be too nervous :p if you use `find_library` / `find_path` it should just work, and you can check that it's picking up the spack search paths by setting `-DCMAKE_FIND_DEBUG_MODE=ON` temporarily.